serial_unix.go

```diff
@@ -87,14 +87,19 @@ func (port *unixPort) Read(p []byte) (n int, err error) {
 		return 0, &PortError{code: PortClosed}
 	}
 
-	fds := unixutils.NewFDSet(port.handle, port.closeSignal.ReadFD())
-	res, err := unixutils.Select(fds, nil, fds, -1)
+	closeSignalFD := port.closeSignal.ReadFD()
+	fds := unixutils.NewFDSet(port.handle, closeSignalFD)
+	var selectResult *unixutils.FDResultSets
+	for repeat := true; repeat; repeat = err == unix.EINTR {
+		selectResult, err = unixutils.Select(fds, nil, fds, -1)
+	}
 	if err != nil {
```

Comment:
How is err in scope here? It seems like it's only declared in the for loop so shouldn't exist at this point.

Reply:
Oh, it's the named return. Not a fan of using that if you're also going to use an explicit return at the end.

Reply:
Agreed, not a fan as well.